### PR TITLE
Fix timezone for Locking creation date

### PR DESCRIPTION
**Pull request by iFlameing**

### CODE REVIEW

File: src/plone/restapi/services/locking/__init__.py [hunk 1]
```diff
@@ -16,7 +16,11 @@ def creator_url(username):
 
 
 def creation_date(timestamp):
-    return datetime.fromtimestamp(timestamp).isoformat()
+    # Note that isoformat does not add the Z at the end of the string, The lack of the
+    # timezone specifier causes Intl (and derivative libraries) in the browser not to
+    # use local time, which is considered a bug in applications. Therefore we add the Z
+    # to the end to make sure that the date will be interpreted properly by the client.
+    return datetime.fromtimestamp(timestamp).strftime("%Y-%m-%dT%H:%M:%SZ")
```
Comment (davisagli) [inline src/plone/restapi/services/locking/__init__.py:23]: Timestamp is a lock modification timestamp which was created in the local timezone here: https://github.com/zopefoundation/Zope/blob/master/src/OFS/LockItem.py#L95

To get it in UTC, we need to use utcfromtimestamp here:
```suggestion
    return datetime.utcfromtimestamp(timestamp).strftime("%Y-%m-%dT%H:%M:%SZ")
```

**Review by davisagli — CHANGES_REQUESTED** [review]
There are also failing tests. The snapshot of the response for this service needs to be updated to include Z

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by reebalazs** [comment]
@iFlameing tests fixed, please proceed as needed.

**Review by jensens — APPROVED** [review]

**Pull request merged**